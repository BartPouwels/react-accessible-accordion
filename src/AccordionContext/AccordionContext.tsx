// tslint:disable:max-classes-per-file

import * as React from 'react';
import AccordionStore from '../AccordionStore/AccordionStore';
import { UUID } from '../ItemContext/ItemContext';

export interface ProviderProps {
    preExpanded?: UUID[];
    allowMultipleExpanded?: boolean;
    allowZeroExpanded?: boolean;
    children?: React.ReactNode;
    onChange?(args: UUID[]): void;
}

type ProviderState = AccordionStore;

export interface AccordionContext {
    allowMultipleExpanded: boolean;
    allowZeroExpanded: boolean;
    setExpanded(uuid: UUID, expanded: boolean): void;
    isItemDisabled(uuid: UUID): boolean;
    isItemExpanded(uuid: UUID): boolean;
}

const Context = React.createContext(null as AccordionContext | null);

export class Provider extends React.PureComponent<
    ProviderProps,
    ProviderState
> {
    static defaultProps: ProviderProps = {
        allowMultipleExpanded: false,
        allowZeroExpanded: false,
    };

    state: ProviderState = new AccordionStore({
        expanded: this.props.preExpanded,
        allowMultipleExpanded: this.props.allowMultipleExpanded,
        allowZeroExpanded: this.props.allowZeroExpanded,
    });

    setExpanded = (key: UUID, expanded: boolean): void => {
        this.setState(
            (state: Readonly<ProviderState>) =>
                state.setExpanded(key, expanded),
            () => {
                if (this.props.onChange) {
                    this.props.onChange(this.state.expanded);
                }
            },
        );
    };

    isItemDisabled = (key: UUID): boolean => {
        return this.state.isItemDisabled(key);
    };

    isItemExpanded = (key: UUID): boolean => {
        return this.state.isItemExpanded(key);
    };

    render(): JSX.Element {
        const { allowZeroExpanded, allowMultipleExpanded } = this.state;

        return (
            <Context.Provider
                value={{
                    allowMultipleExpanded,
                    allowZeroExpanded,
                    setExpanded: this.setExpanded,
                    isItemDisabled: this.isItemDisabled,
                    isItemExpanded: this.isItemExpanded,
                }}
            >
                {this.props.children || null}
            </Context.Provider>
        );
    }
}

export class Consumer extends React.PureComponent<{
    children(container: AccordionContext): React.ReactNode;
}> {
    renderChildren = (container: AccordionContext | null): React.ReactNode => {
        return container ? this.props.children(container) : null;
    };

    render(): JSX.Element {
        return <Context.Consumer>{this.renderChildren}</Context.Consumer>;
    }
}
