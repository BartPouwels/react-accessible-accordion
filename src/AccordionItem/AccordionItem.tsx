import { default as classnames } from 'classnames';
import * as React from 'react';
import { AccordionContainer } from '../AccordionContainer/AccordionContainer';
import { UUID } from '../ItemContainer/ItemContainer';

type AccordionItemProps = React.HTMLAttributes<HTMLDivElement> & {
    uuid: UUID;
    hideBodyClassName?: string;
    disabled?: boolean;
    expanded?: boolean;
    accordionStore: AccordionContainer;
};

class AccordionItem extends React.Component<AccordionItemProps> {
    componentDidMount(): void {
        const { uuid, accordionStore, disabled } = this.props;

        accordionStore.addItem({
            uuid,
            expanded: this.props.expanded || false,
            disabled,
        });
    }

    componentWillUnmount(): void {
        this.props.accordionStore.removeItem(this.props.uuid);
    }

    // This is here so that the user can dynamically set the 'expanded' state using the 'expanded' prop.
    componentDidUpdate(prevProps: AccordionItemProps): void {
        const { uuid, expanded, accordionStore } = this.props;
        if (expanded !== prevProps.expanded) {
            accordionStore.setExpanded(uuid, expanded);
        }
    }

    render(): JSX.Element {
        const {
            uuid,
            className,
            hideBodyClassName,
            accordionStore,
            disabled,
            expanded,
            ...rest
        } = this.props;

        // Deliberately not using 'find' because IE compat.
        const currentItem = accordionStore.items.filter(
            item => item.uuid === uuid,
        )[0];

        if (!currentItem) {
            return null;
        }

        return (
            <div
                className={classnames(className, {
                    [String(hideBodyClassName)]:
                        !currentItem.expanded && hideBodyClassName,
                })}
                {...rest}
            />
        );
    }
}

export default AccordionItem;
