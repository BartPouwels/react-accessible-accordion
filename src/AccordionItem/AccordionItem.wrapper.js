// @flow

import React, { Component } from 'react';
import type { ElementProps } from 'react';

import { Provider, Subscribe } from 'unstated';
import {
    Consumer,
    type AccordionContainer,
} from '../AccordionContainer/AccordionContainer';
import ItemContainer from '../ItemContainer/ItemContainer';
import AccordionItem from './AccordionItem';

type AccordionItemWrapperProps = ElementProps<'div'> & {
    hideBodyClassName: ?string,
    disabled: ?boolean,
    expanded: ?boolean,
    // accordionStore: AccordionContainer,
    uuid?: string,
};

const defaultProps = {
    className: 'accordion__item',
    hideBodyClassName: '',
    disabled: false,
    expanded: false,
    uuid: undefined,
};

class AccordionItemWrapper extends Component<AccordionItemWrapperProps> {
    itemContainer = new ItemContainer({
        uuid: this.props.uuid,
    });

    static defaultProps = defaultProps;

    accordionContainer: AccordionContainer;

    renderAccordionChildren = (accordionStore: AccordionContainer) => {
        this.accordionContainer = accordionStore;
        return (
            <Provider inject={[this.itemContainer]}>
                <Subscribe to={[ItemContainer]}>
                    {this.renderItemChildren}
                </Subscribe>
            </Provider>
        );
    };

    renderItemChildren = (itemStore: ItemContainer) => {
        const { uuid } = itemStore.state;
        return (
            <AccordionItem
                {...this.props}
                uuid={uuid}
                accordionStore={this.accordionContainer}
            />
        );
    };

    render() {
        return <Consumer>{this.renderAccordionChildren}</Consumer>;
    }
}

export default AccordionItemWrapper;
