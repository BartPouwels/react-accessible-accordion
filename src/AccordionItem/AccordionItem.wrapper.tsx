import consecutive from 'consecutive';
import * as React from 'react';

import {
    CONTEXT_KEY,
    getAccordionStore,
} from '../AccordionContainer/AccordionContainer';
import { Provider as ItemProvider } from '../ItemContainer/ItemContainer';
import AccordionItem from './AccordionItem';

type AccordionItemWrapperProps = React.HTMLAttributes<HTMLDivElement> & {
    hideBodyClassName?: string;
    disabled?: boolean;
    expanded?: boolean;
    uuid?: string;
};

let nextUuid = consecutive();
export function resetNextUuid() {
    nextUuid = consecutive();
}

export default class AccordionItemWrapper extends React.Component<
    AccordionItemWrapperProps
> {
    static contextTypes = {
        // Empty anonymous callback is a hacky 'wildcard' workaround for bypassing prop-types.
        [CONTEXT_KEY]: () => null,
    };

    static defaultProps = {
        className: 'accordion__item',
        hideBodyClassName: '',
        disabled: false,
        expanded: false,
        uuid: undefined,
    };

    id = nextUuid();

    render() {
        const accordionStore = getAccordionStore(this.context);
        if (!accordionStore) {
            // TODO: log an error/warning?
            return null;
        }
        const { uuid, ...rest } = this.props;
        const itemUuid = uuid !== undefined ? uuid : this.id;

        return (
            <ItemProvider uuid={itemUuid}>
                <AccordionItem
                    {...rest}
                    uuid={itemUuid}
                    accordionStore={accordionStore}
                />
            </ItemProvider>
        );
    }
}
