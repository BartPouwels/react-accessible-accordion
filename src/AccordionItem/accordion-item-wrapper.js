// @flow

import React, { Component } from 'react';
import type { ElementProps } from 'react';

import { Provider, Subscribe } from 'unstated';
import AccordionContainer from '../AccordionContainer/AccordionContainer';
import ItemContainer from '../ItemContainer/ItemContainer';
import AccordionItem from './accordion-item';

type AccordionItemWrapperProps = ElementProps<'div'> & {
    hideBodyClassName: ?string,
    disabled: ?boolean,
    expanded: ?boolean,
    accordionStore: AccordionContainer,
    uuid?: string,
};

const defaultProps = {
    className: 'accordion__item',
    hideBodyClassName: '',
    disabled: false,
    expanded: false,
    accordionStore: new AccordionContainer(),
    uuid: null,
};

class AccordionItemWrapper extends Component<AccordionItemWrapperProps> {
    itemContainer = new ItemContainer();
    static defaultProps = defaultProps;

    render() {
        return (
            <Provider inject={[this.itemContainer]}>
                <Subscribe to={[AccordionContainer, ItemContainer]}>
                    {(accordionStore, itemStore) => {
                        let uuid = itemStore.state.uuid;
                        if (this.props.uuid) uuid = this.props.uuid;
                        return (
                            <AccordionItem
                                {...this.props}
                                uuid={uuid}
                                accordionStore={accordionStore}
                                itemstore={itemStore}
                            />
                        );
                    }}
                </Subscribe>
            </Provider>
        );
    }
}

export default AccordionItemWrapper;
