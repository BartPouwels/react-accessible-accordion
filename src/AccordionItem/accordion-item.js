// @flow

import React, { Component } from 'react';
import type { ElementProps } from 'react';

import classNames from 'classnames';
import AccordionContainer from '../AccordionContainer/AccordionContainer';
import ItemContainer from '../ItemContainer/ItemContainer';

type AccordionItemProps = ElementProps<'div'> & {
    uuid: string | number,
    hideBodyClassName: ?string,
    disabled: ?boolean,
    expanded: ?boolean,
    accordionStore: AccordionContainer,
    itemStore: ItemContainer,
};

class AccordionItem extends Component<AccordionItemProps, *> {
    componentWillMount() {
        const { uuid, accordionStore, itemStore, disabled } = this.props;

        itemStore.setUuid(uuid);

        const currentItem = accordionStore.state.items.find(
            item => item.uuid === uuid,
        );
        if (currentItem)
            // eslint-disable-next-line no-console
            console.error(
                `AccordionItem error: One item already has the uuid "${uuid}". Uuid property must be unique. See: https://github.com/springload/react-accessible-accordion#accordionitem`,
            );

        accordionStore.addItem({
            uuid,
            expanded: this.props.expanded || false,
            disabled,
        });
    }

    componentWillUnmount() {
        this.props.accordionStore.removeItem(this.props.uuid);
    }

    // This is here so that the user can dynamically set the 'expanded' state using the 'expanded' prop.
    componentWillReceiveProps({
        uuid,
        expanded,
        accordionStore,
    }: AccordionItemProps) {
        if (expanded !== this.props.expanded) {
            accordionStore.setExpanded(uuid, expanded);
        }
    }

    render() {
        const {
            uuid,
            className,
            hideBodyClassName,
            accordionStore,
            disabled,
            expanded,
            itemStore,
            ...rest
        } = this.props;

        const currentItem = accordionStore.state.items.find(
            item => item.uuid === uuid,
        );

        if (!currentItem) {
            return null;
        }

        return (
            <div
                className={classNames(className, {
                    [hideBodyClassName]:
                        !currentItem.expanded && hideBodyClassName,
                })}
                {...rest}
            />
        );
    }
}

export default AccordionItem;
