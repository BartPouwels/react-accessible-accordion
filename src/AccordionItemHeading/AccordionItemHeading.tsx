import { default as classnames } from 'classnames';
import * as React from 'react';
import {
    focusFirstSiblingOf,
    focusLastSiblingOf,
    focusNextSiblingOf,
    focusPreviousSiblingOf,
} from '../helpers/focus';
import keycodes from '../helpers/keycodes';

import { DivAttributes } from '../helpers/types';
import {
    Consumer as ItemConsumer,
    ItemContext,
} from '../ItemContext/ItemContext';

type Props = Pick<DivAttributes, Exclude<keyof DivAttributes, 'role'>> & {
    expandedClassName: string;
    expanded: boolean;
    toggleExpanded(): void;
};

export class AccordionItemHeading extends React.PureComponent<Props> {
    handleKeyPress = (evt: React.KeyboardEvent<HTMLDivElement>): void => {
        const keyCode = evt.which.toString();

        if (keyCode === keycodes.ENTER || keyCode === keycodes.SPACE) {
            evt.preventDefault();
            this.props.toggleExpanded();
        }

        if (evt.target instanceof HTMLElement) {
            switch (keyCode) {
                case keycodes.HOME: {
                    evt.preventDefault();
                    focusFirstSiblingOf(evt.target);
                    break;
                }
                case keycodes.END: {
                    evt.preventDefault();
                    focusLastSiblingOf(evt.target);
                    break;
                }
                case keycodes.LEFT:
                case keycodes.UP: {
                    evt.preventDefault();
                    focusPreviousSiblingOf(evt.target);
                    break;
                }
                case keycodes.RIGHT:
                case keycodes.DOWN: {
                    evt.preventDefault();
                    focusNextSiblingOf(evt.target);
                    break;
                }
                default: {
                    //
                }
            }
        }
    };

    render(): JSX.Element {
        const {
            className,
            expandedClassName,
            expanded,
            toggleExpanded,
            ...rest
        } = this.props;

        const headingClassName = classnames(className, {
            [expandedClassName]: expandedClassName && expanded,
        });

        return (
            <div
                // tslint:disable-next-line react-a11y-event-has-role
                className={headingClassName}
                onClick={toggleExpanded}
                data-accordion-component="AccordionItemHeading"
                onKeyDown={this.handleKeyPress}
                {...rest}
            />
        );
    }
}

type WrapperProps = Pick<
    Props,
    Exclude<keyof Props, 'toggleExpanded' | 'expanded'>
>;

const Wrapper: React.SFC<WrapperProps> = (props: WrapperProps): JSX.Element => (
    <ItemConsumer>
        {(itemContext: ItemContext): JSX.Element => {
            const { expanded, toggleExpanded, headingAttributes } = itemContext;

            return (
                <AccordionItemHeading
                    {...props}
                    expanded={expanded}
                    toggleExpanded={toggleExpanded}
                    {...headingAttributes}
                />
            );
        }}
    </ItemConsumer>
);

export default Wrapper;
