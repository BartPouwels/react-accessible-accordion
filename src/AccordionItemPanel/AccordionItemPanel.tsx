import { default as classnames } from 'classnames';
import * as React from 'react';
import { UUID } from '../ItemContainer/ItemContainer';

type AccordionItemPanelProps = React.HTMLAttributes<HTMLDivElement> & {
    hideBodyClassName: string;
    uuid: UUID;
    expanded: boolean;
    disabled: boolean;
    accordion: boolean;
};

const AccordionItemPanel = (props: AccordionItemPanelProps): JSX.Element => {
    const {
        className,
        hideBodyClassName,
        uuid,
        expanded,
        disabled,
        accordion,
        ...rest
    } = props;

    const role = accordion ? 'region' : null;
    const hideAriaAttribute = accordion ? null : !expanded;

    return (
        <div
            id={`accordion__panel-${uuid}`}
            className={classnames(className, {
                [hideBodyClassName]: !expanded,
            })}
            aria-hidden={hideAriaAttribute}
            aria-labelledby={`accordion__heading-${uuid}`}
            role={role}
            {...rest}
        />
    );
};

export default AccordionItemPanel;
