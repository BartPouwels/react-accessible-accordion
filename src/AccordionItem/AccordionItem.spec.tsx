import { mount, ReactWrapper } from 'enzyme';
import * as React from 'react';
import { Provider as AccordionProvider } from '../AccordionContext/AccordionContext';
import { default as AccordionItemHeading } from '../AccordionItemHeading/AccordionItemHeading.wrapper';
import { default as AccordionItemPanel } from '../AccordionItemPanel/AccordionItemPanel.wrapper';
import { resetNextUuid } from '../helpers/uuid';
import { Provider as ItemProvider, UUID } from '../ItemContext/ItemContext';
import { default as AccordionItem } from './AccordionItem.wrapper';

describe('AccordionItem', () => {
    beforeEach(() => {
        resetNextUuid();
    });

    it('renders null outside the context of an accordion', () => {
        const wrapper = mount(<AccordionItem />);
        expect(wrapper.html()).toBeNull();
    });

    it('renders correctly with allowMultipleExpanded false', () => {
        const wrapper = mount(
            <AccordionProvider>
                <AccordionItem className="accordion__item">
                    <AccordionItemHeading className="accordion__heading">
                        <div>Fake title</div>
                    </AccordionItemHeading>
                    <AccordionItemPanel className="accordion__panel">
                        <div>Fake body</div>
                    </AccordionItemPanel>
                </AccordionItem>
            </AccordionProvider>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with allowMultipleExpanded true', () => {
        const wrapper = mount(
            <AccordionProvider allowMultipleExpanded={true}>
                <AccordionItem className="accordion__item">
                    <AccordionItemHeading className="accordion__heading">
                        <div>Fake title</div>
                    </AccordionItemHeading>
                    <AccordionItemPanel className="accordion__panel">
                        <div>Fake body</div>
                    </AccordionItemPanel>
                </AccordionItem>
            </AccordionProvider>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with allowZeroExpanded false', () => {
        const wrapper = mount(
            <AccordionProvider>
                <AccordionItem className="accordion__item">
                    <AccordionItemHeading className="accordion__heading">
                        <div>Fake title</div>
                    </AccordionItemHeading>
                    <AccordionItemPanel className="accordion__panel">
                        <div>Fake body</div>
                    </AccordionItemPanel>
                </AccordionItem>
            </AccordionProvider>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with allowZeroExpanded true', () => {
        const wrapper = mount(
            <AccordionProvider allowZeroExpanded={true}>
                <AccordionItem className="accordion__item">
                    <AccordionItemHeading className="accordion__heading">
                        <div>Fake title</div>
                    </AccordionItemHeading>
                    <AccordionItemPanel className="accordion__panel">
                        <div>Fake body</div>
                    </AccordionItemPanel>
                </AccordionItem>
            </AccordionProvider>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders with multiple AccordionItems', () => {
        const wrapper = mount(
            <AccordionProvider>
                <div>
                    <AccordionItem>
                        <div>Fake title</div>
                        <div>Fake body</div>
                    </AccordionItem>
                    <AccordionItem>
                        <div>Fake title</div>
                        <div>Fake body</div>
                    </AccordionItem>
                </div>
            </AccordionProvider>,
        );

        expect(wrapper.find(AccordionItem).length).toEqual(2);
    });

    it('still renders with no AccordionItemHeading or AccordionItemPanel', () => {
        const wrapper = mount(
            <AccordionProvider>
                <AccordionItem>
                    <div>Fake title</div>
                    <div>Fake body</div>
                </AccordionItem>
            </AccordionProvider>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('still renders with no children at all', () => {
        const wrapper = mount(
            <AccordionProvider>
                <AccordionItem />
            </AccordionProvider>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders with different className', () => {
        const wrapper = mount(
            <AccordionProvider>
                <AccordionItem className="testCSSClass" />
            </AccordionProvider>,
        );

        expect(wrapper.find('div').prop('className')).toBe('testCSSClass');
    });

    it('renders with different expandedClassName', () => {
        const uuid = 'foo';
        const wrapper = mount(
            <AccordionProvider preExpanded={[uuid]}>
                <AccordionItem
                    uuid={uuid}
                    className="AccordionItem"
                    expandedClassName="AccordionItem--expanded"
                />
            </AccordionProvider>,
        );

        expect(wrapper.find('div').prop('className')).toEqual(
            'AccordionItem AccordionItem--expanded',
        );
    });

    it('renders correctly with other blocks inside', () => {
        const wrapper = mount(
            <AccordionProvider>
                <AccordionItem>
                    <AccordionItemHeading>
                        <div>Fake title</div>
                    </AccordionItemHeading>
                    <div>Just another block</div>
                    <AccordionItemPanel>
                        <div>Fake body</div>
                    </AccordionItemPanel>
                </AccordionItem>
            </AccordionProvider>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with other blocks inside 2', () => {
        const wrapper = mount(
            <AccordionProvider>
                <AccordionItem>
                    <AccordionItemHeading>
                        <div>Fake title</div>
                    </AccordionItemHeading>
                    <AccordionItemPanel>
                        <div>Fake body</div>
                    </AccordionItemPanel>
                    <div>Just another block</div>
                </AccordionItem>
            </AccordionProvider>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('dynamically changing arbitrary props does not affect expanded state', () => {
        const Wrapper = ({ className }: { className: string }): JSX.Element => (
            <AccordionProvider>
                <AccordionItem className={className}>
                    <AccordionItemHeading>
                        <div>Fake title</div>
                    </AccordionItemHeading>
                </AccordionItem>
            </AccordionProvider>
        );
        const wrapper = mount(<Wrapper className="foo" />);
        const instance = wrapper
            .find(AccordionProvider)
            .instance() as AccordionProvider;

        wrapper.setProps({ className: 'bar' });

        expect(instance.state.expanded.length).toEqual(0);
    });

    it('respects arbitrary user-defined props', () => {
        const wrapper = mount(
            <AccordionProvider>
                <AccordionItem lang="en" />
            </AccordionProvider>,
        );

        const div = wrapper.find('div').getDOMNode();

        expect(div.getAttribute('lang')).toEqual('en');
    });

    it('generates unique uuids', () => {
        const wrapper = mount(
            <AccordionProvider>
                <AccordionItem />
                <AccordionItem />
            </AccordionProvider>,
        );

        const uuids = wrapper
            .find(ItemProvider)
            .map((provider: ReactWrapper) => provider.prop('uuid'));

        expect(uuids.length).toEqual(2);
        expect(uuids[0]).not.toEqual(uuids[1]);
    });
});
