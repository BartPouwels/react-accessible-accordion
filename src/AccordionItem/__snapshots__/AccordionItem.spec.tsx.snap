// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItem renders correctly with accordion false 1`] = `
<div
  className="accordion__item"
>
  <div
    aria-controls="accordion__body-0"
    aria-expanded={false}
    className="accordion__title"
    id="accordion__title-0"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div>
      Fake title
    </div>
  </div>
  <div
    aria-hidden={true}
    aria-labelledby="accordion__title-0"
    className="accordion__body accordion__body--hidden"
    id="accordion__body-0"
    role={null}
  >
    <div>
      Fake body
    </div>
  </div>
</div>
`;

exports[`AccordionItem renders correctly with accordion true 1`] = `
<div
  className="accordion__item"
>
  <div
    aria-controls="accordion__body-0"
    aria-expanded={false}
    className="accordion__title"
    id="accordion__title-0"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div>
      Fake title
    </div>
  </div>
  <div
    aria-hidden={null}
    aria-labelledby="accordion__title-0"
    className="accordion__body accordion__body--hidden"
    id="accordion__body-0"
    role="region"
  >
    <div>
      Fake body
    </div>
  </div>
</div>
`;

exports[`AccordionItem renders correctly with other blocks inside 1`] = `
<div
  className="accordion__item"
>
  <div
    aria-controls="accordion__body-0"
    aria-expanded={false}
    className="accordion__title"
    id="accordion__title-0"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div>
      Fake title
    </div>
  </div>
  <div>
    Just another block
  </div>
  <div
    aria-hidden={true}
    aria-labelledby="accordion__title-0"
    className="accordion__body accordion__body--hidden"
    id="accordion__body-0"
    role={null}
  >
    <div>
      Fake body
    </div>
  </div>
</div>
`;

exports[`AccordionItem renders correctly with other blocks inside 2 1`] = `
<div
  className="accordion__item"
>
  <div
    aria-controls="accordion__body-0"
    aria-expanded={false}
    className="accordion__title"
    id="accordion__title-0"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div>
      Fake title
    </div>
  </div>
  <div
    aria-hidden={true}
    aria-labelledby="accordion__title-0"
    className="accordion__body accordion__body--hidden"
    id="accordion__body-0"
    role={null}
  >
    <div>
      Fake body
    </div>
  </div>
  <div>
    Just another block
  </div>
</div>
`;

exports[`AccordionItem still renders with no AccordionItemTitle or AccordionItemBody 1`] = `
<div
  className="accordion__item"
>
  <div>
    Fake title
  </div>
  <div>
    Fake body
  </div>
</div>
`;

exports[`AccordionItem still renders with no children at all 1`] = `
<div
  className="accordion__item"
/>
`;
